src/Couchbase.Lite.Tests.Shared/ReplicationTest.cs

```diff
@@ -1339,24 +1338,36 @@ public void TestNonBlockingDatabaseOperationConflictResolver()
                 if (resolveCnt == 0) {
                     using (var d = Db.GetDocument("doc1"))
                     using (var doc = d.ToMutable()) {
-                        d.GetString("name").Should().Be("Tiger");
+                        d.GetString("name").Should().Be("Cat");
                         doc.SetString("name", "Cougar");
                         Db.Save(doc);
-                        d.GetString("name").Should().Be("Cougar", "Because database save operation was not blocked");
+                        doc.GetString("name").Should().Be("Cougar", "Because database save operation was not blocked");
                     }
                 }
                 resolveCnt++;
                 return null;
             });
 
-            RunReplication(config, 0, 0);
+            WaitAssert _waitAssert1 = new WaitAssert();
+            RunReplication(config, 0, 0, documentReplicated: (sender, args) =>
+            {
+                if (!args.IsPush) {
+                    foreach (var d in args.Documents) {
+                        _waitAssert1.RunConditionalAssert(() =>
+                        {
+                            return d.Error == null;
+                        });
+                    }
+                }
+            });
 
+            _waitAssert1.WaitForResult(TimeSpan.FromSeconds(2));
```

Comment:
This will work, but it won't tell you much about the error.  Remember that these need to be maintained and explained to the next person who works on them so simpler is better.  Perhaps something like:

```
Exception innerException = null;
config.ConflictResolver = new TestConflictResolver((conflict) =>
{
    try {
        // ...
    } catch(Exception e) {
        innerException = e;
    }
});

innerException.Should.BeNull();
```

Or in this case, there is something even simpler you can do.  You can ignore the above, let the exception throw, and notice that `resolveCnt` does not change in that case because it's below the part that throws the exception.  So by that logic, the last part of this should be:

```
// This will be 0 if the test resolver threw an exception
resolveCnt.Should().NotBe(0, "because otherwise the conflict resolver didn't complete"); 
using (var doc = Db.GetDocument("doc1")) {
    // Remove the if here, it doesn't add anything (and it's incorrect)
    doc.Should().BeNull();
}
```